NCS.DSS.Customer/AzureSearchDataSyncTrigger/CustomerSearchDataSyncTrigger.cs

```diff
@@ -1,91 +1,90 @@
 
-            _loggerHelper.LogInformationMessage(_log, correlationId, "Documents modified " + documents.Count);
+            _logger.LogInformation("Documents modified {count}",documents.Count);
 
             if (documents.Count > 0)
-            {
-                var customers = documents.Select(doc => new Models.CustomerSearch()
-                {
-                    CustomerId = doc.GetPropertyValue<Guid?>("id"),
-                    DateOfRegistration = doc.GetPropertyValue<DateTime?>("DateOfRegistration"),
-                    Title = doc.GetPropertyValue<Title>("Title"),
-                    GivenName = doc.GetPropertyValue<string>("GivenName"),
-                    FamilyName = doc.GetPropertyValue<string>("FamilyName"),
-                    DateofBirth = doc.GetPropertyValue<DateTime?>("DateofBirth"),
-                    Gender = doc.GetPropertyValue<Gender?>("Gender"),
-                    UniqueLearnerNumber = doc.GetPropertyValue<string>("UniqueLearnerNumber"),
-                    OptInUserResearch = doc.GetPropertyValue<bool?>("OptInUserResearch"),
-                    OptInMarketResearch = doc.GetPropertyValue<bool?>("OptInMarketResearch"),
-                    DateOfTermination = doc.GetPropertyValue<DateTime?>("DateOfTermination"),
-                    ReasonForTermination = doc.GetPropertyValue<ReasonForTermination?>("ReasonForTermination"),
-                    IntroducedBy = doc.GetPropertyValue<IntroducedBy?>("IntroducedBy"),
-                    IntroducedByAdditionalInfo = doc.GetPropertyValue<string>("IntroducedByAdditionalInfo"),
-                    LastModifiedDate = doc.GetPropertyValue<DateTime?>("LastModifiedDate"),
-                    LastModifiedTouchpointId = doc.GetPropertyValue<string>("LastModifiedTouchpointId")
-                })
-                    .ToList();
-
-                var batch = IndexDocumentsBatch.MergeOrUpload(customers);
+            { 
+                try
+                {      
+                    var customers = new List<Models.CustomerSearch>();
+                    foreach (var doc in documents)
+                    {
+                        var custJson = JsonSerializer.Serialize(doc);
+                        var customer = JsonSerializer.Deserialize<Models.CustomerSearch>(custJson);
+                        if(doc.CustomerId == null && doc.id != null)
+                            customer.CustomerId = doc.id;
+                        else if(doc.id == null && doc.CustomerId != null)
+                            customer.CustomerId = doc.CustomerId;
+
+                        customers.Add(customer);
+                    }
+                    var custFiltered = customers.Where(d => d.CustomerId != null);
+                    if (custFiltered.Any())
+                    {
+                        var batch = IndexDocumentsBatch.MergeOrUpload(custFiltered);
 
+                        _logger.LogInformation("attempting to merge docs to azure search");
```

Comment:
Can we capitalise the A please, also can you specify what "docs" are being merged and not use abbrieviations